File: src/algorithms/strings/palindrome-check/solution2.go
package main

// O(n) time | O(n) space
func IsPalindrome(str string) bool {
	result := []byte{}
	for i := len(str) - 1; i >= 0; i-- {
		result = append(result, str[i])
	}
	return str == string(result)
}

File: src/algorithms/strings/palindrome-check/solution2.js
// O(n) time | O(n) space
function isPalindrome(string) {
	const reversedChars = [];
	for (let i = string.length - 1; i >= 0; i--) {
		reversedChars.push(string[i]);
	}
	return string === reversedChars.join('');
};

exports.isPalindrome = isPalindrome;